src/tracers/blob_saver_tracer.rs

```diff
@@ -0,0 +1,102 @@
+use super::tracer::Tracer;
+use crate::{eravm_error::EraVmError, state::VMState, value::FatPointer, Opcode};
+use std::collections::HashMap;
+use u256::{H160, H256, U256};
+use zkevm_opcode_defs::ethereum_types::Address;
+
+#[derive(Default, Debug)]
+pub struct BlobSaverTracer {
+    pub blobs: HashMap<U256, Vec<U256>>,
+}
+
+impl BlobSaverTracer {
+    pub fn new() -> Self {
+        Self {
+            blobs: Default::default(),
+        }
+    }
+}
+
+const CONTRACT_DEPLOYER_ADDRESS: Address = H160([
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+    0x00, 0x00, 0x80, 0x06,
+]);
+
+const KNOWN_CODES_STORAGE_ADDRESS: Address = H160([
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+    0x00, 0x00, 0x80, 0x04,
+]);
+
+pub(crate) fn hash_evm_bytecode(bytecode: &[u8]) -> H256 {
+    use zkevm_opcode_defs::sha2::{Digest, Sha256};
+    let mut hasher = Sha256::new();
+    let len = bytecode.len() as u16;
+    hasher.update(bytecode);
+    let result = hasher.finalize();
+
+    let mut output = [0u8; 32];
+    output[..].copy_from_slice(result.as_slice());
+    output[0] = 2; //BLOB version byte
+    output[1] = 0;
+    output[2..4].copy_from_slice(&len.to_be_bytes());
+
+    H256(output)
+}
+
+fn bytes_to_be_words(vec: &[u8]) -> Vec<U256> {
+    assert!(vec.len() % 32 == 0, "Invalid bytecode length");
```

Comment:
This will panic if `vec.len() % 32 != 0`, let's make it return an error if that happens.

Reply:
I modified the code a little bit since you made this comment. Now there is a check but we are not returning an error if the bytecode len is not multiple of 32. Instead, we return an `Ok(())` and log an error. Do you think it's better return an error from the tracer and stop execution?